src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -22,4 +24,29 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object rectangle) {
+        if (rectangle == this) {
+            return true;
+        }
+        if (rectangle == null) {
+            return false;
+        }
+        if (rectangle.getClass().equals(Rectangle.class)) {
+            Rectangle rectangle1 = (Rectangle) rectangle;
+            return Objects.equals(getColor(), rectangle1.color)
+                    && Objects.equals(getLength(), rectangle1.length)
+                    && Objects.equals(getWidth(), rectangle1.width);
```

Comment:
https://mate-academy.github.io/jv-program-common-mistakes/java-core/equals-and-hashcode/equals-and-hashcode